lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp

```diff
@@ -94,12 +107,37 @@ void OpenflowController::dispatch_event(const ControllerEvent& ev) {
 void OpenflowController::inject_external_event(
     std::shared_ptr<ExternalEvent> ev, void* (*cb)(std::shared_ptr<void>) ) {
   if (latest_ofconn_ == NULL) {
-    OAILOG_ERROR(
-        LOG_GTPV1U, "Null connection on event type %d", ev->get_type());
-    throw std::runtime_error("Controller not connected to switch:\n");
+#define CONNECTION_EVENT_WAIT_TIME 5
+    if (!is_controller_connected_to_switch(CONNECTION_EVENT_WAIT_TIME)) {
```

Comment:
This appears as if we are checking for the failure of `is_controller_connected_to_switch`, instead can we do
```suggestion
    if (is_controller_connected_to_switch(CONNECTION_EVENT_WAIT_TIME) == RETURNok) {
```